geetools/ee_feature_collection.py

```diff
@@ -35,6 +35,46 @@ def toImage(
         width == "" or params.update(width=width)
         return ee.Image().paint(self._obj, **params)
 
+    def toDictionary(
+        self, keyColumn: str | ee.String = "system:index", selectors: list | ee.List = []
+    ) -> ee.Dictionary:
+        """Convert to Dictionary.
+
+        Parameters:
+            keyColumn: the column to use as keys. Must contain unique values, if not it will fail.
+            selectors: a list of properties to add in the output. If the list is empty all properties will be added.
+
+        Returns:
+            a ee.Dictionary with values of keyColumn as keys and ee.Dictionary as
+            values. The output will look like:
+
+            {
+             '00000000000000000010': {'ADM0_CODE': 74578, 'ADM0_NAME': 'Azores Islands'},
+             '00000000000000000011': {'ADM0_CODE': 7, 'ADM0_NAME': 'Andorra'}
+            }
+        """
+        uniqueIds = self._obj.aggregate_array(keyColumn)
+        selectors = ee.List(selectors)
+        keyColumn = ee.String(keyColumn)
+        finalSelectors = ee.List(
+            ee.Algorithms.If(
+                selectors.size().eq(0), self._obj.first().propertyNames(), selectors  # all columns
+            )
+        )
```

Comment:
There are 2 (different) comments here: 
1. mutable default param
2. avoid `ee.Algorithms.If`

1) Actually, I've seen that you use it all the time, so I considered you preferred this way (once you explicitly told me so). But I totally agree with Nataliia, which is actually the same as PyCharm suggests. I see 2 ways to avoid it:
a. use a `tuple` and convert to `list` inside the method
b. use `Optional[list] = None` (the one suggested in the post).

2) I did it this way to be able to use the method in a server-side function (e.g. `map`). Also, if the user passes an empty `ee.List` the conditional checking on the client-side will pass as `True`.

Reply:
ok sorry for the double information but now I remember why we can actually use the `ee.List | list = []` in `geetools`. using a mutable object in the default parameter is a problem if you modify it afterward but we here never do, we simply check the length and if there is nothing we use another list. so it's tricky but perfectly legit in the context of a GEE casting ;-)